### PR TITLE
[Add Project] JSQDataSourcesKit

### PR DESCRIPTION
**Pull request by jessesquires**
### Pull Request Description

This adds my project, [JSQDataSourcesKit](https://github.com/jessesquires/JSQDataSourcesKit).

**NOTE:** This project has revealed a regression in Swift 4. See: [SR-4088](https://bugs.swift.org/browse/SR-4088).

### Acceptance Criteria

To be accepted into the Swift source compatibility test suite, a project must:

- [x] be an *Xcode* or *swift package manager* project
- [x] support building on either Linux or macOS
- [x] target Linux, macOS, or iOS/tvOS/watchOS device
- [x] be contained in a publicly accessible git repository
- [x] maintain a project branch that builds against Swift 3.0 compatibility mode
      and passes any unit tests
- [x] have maintainers who will commit to resolve issues in a timely manner
- [x] be compatible with the latest GM/Beta versions of *Xcode* and *swiftpm*
- [x] add value not already included in the suite
- [x] be licensed with one of the following permissive licenses:
	* BSD
	* MIT
	* Apache License, version 2.0
	* Eclipse Public License
	* Mozilla Public License (MPL) 1.1
	* MPL 2.0
	* CDDL
- [ ] pass ./check script run

Ensure project meets all listed requirements before submitting a pull request.

### CODE REVIEW

File: projects.json [hunk 1]
```diff
@@ -2238,5 +2238,35 @@
...
+        "target": "JSQDataSourcesKit-iOS",
+        "destination": "generic/platform=iOS",
+        "configuration": "Release"
+      },
+      {
+        "action": "TestXcodeProjectScheme",
+        "project": "JSQDataSourcesKit.xcodeproj",
+        "scheme": "JSQDataSourcesKitTests",
+        "destination": "platform=iOS Simulator,name=iPhone 6s"
+      },
```
Comment (lplarson) [inline projects.json]: This comma is preventing the runner from loading the project list.

File: projects.json [hunk 2]
```diff
@@ -2238,5 +2238,35 @@
...
+    ]
+  },
```
Comment (lplarson) [inline projects.json]: This comma as well.

**Comment by lplarson** [comment]
@jessesquires Please resolve the mentioned issues. Otherwise, looks good!

File: .gitignore [hunk 3]
```diff
@@ -1,3 +1,4 @@
+.DS_Store
```
Comment (jessesquires) [inline .gitignore]: hope this is ok to include 😄 

(i'm sure other macOS users would need this)

**Comment by jessesquires** [comment]
oops 😳. fixed.

thanks @lplarson ! 💯

**Comment by lplarson** [comment]
@swift-ci test and merge

**Comment by lplarson** [comment]
@swift-ci test

**Comment by lplarson** [comment]
@swift-ci Please test

**Comment by lplarson** [comment]
Thanks. Looks good!

**Comment by jessesquires** [comment]
thanks @lplarson !

**Pull request merged**